### PR TITLE
feat: view and select previously saved shipping address

### PR DESCRIPTION
**Pull request by sss-bct**

### CODE REVIEW

File: components/checkout/ShippingStep/ShippingStep.tsx [hunk 1]
```diff
@@ -113,6 +117,17 @@ const ShippingStep = (props: ShippingProps) => {
       <Typography variant="h2" component="h2" sx={{ fontWeight: 'bold' }}>
         {t('shipping')}
       </Typography>
+      {
+        <AddressList
+          radio={true}
+          heading={t('common:your-default-shipping-address')}
+          subHeading={t('common:previously-saved-shipping-addresses')}
+          addresses={userShippingAddress}
+          radioGroupTitle={'Your default silling dddress'}
```
Comment (kibo-kevinwatts) [inline components/checkout/ShippingStep/ShippingStep.tsx:126]: use translation string
also fix 'silling' typo

File: components/common/AddressList/AddressList.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+Common.args = {
+  withoutRadioTitle: '',
+  addresses: userAddressResponse.items as CustomerContact[],
+}
+
+export const Radio = Template.bind({})
+
+Radio.args = {
+  radio: true,
+  radioGroupTitle: 'Your default silling address',
```
Comment (kibo-kevinwatts) [inline components/common/AddressList/AddressList.stories.tsx:27]: fix typo

File: components/common/AddressList/AddressList.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        <Typography variant="h4" fontWeight={'bold'}>
+          {t('common:no-saved-addresses-yet')}
+        </Typography>
+      )}
+    </Box>
+  )
+}
+
+const KiboAddressList = (props: AddressListProps) => {
+  const { addresses, withoutRadioTitle } = props
```
Comment (kibo-kevinwatts) [inline components/common/AddressList/AddressList.tsx:85]: Why add the extra 'withoutRadioTitle' prop? 
Can we just leave "title" null?

File: hooks/queries/useUserAddresses/useUserAddressQueries.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  const client = makeGraphQLClient()
+  const response = await client.request({
+    document: getUserAddressesQuery,
+    variables: { accountId },
+  })
+  return response?.customerAccountContacts
+}
+
+export const useUserAddressesQueries = ({ accountId }: UseUserAddresses): UserAddressType => {
+  try {
```
Comment (kibo-kevinwatts) [inline hooks/queries/useUserAddresses/useUserAddressQueries.ts:28]: why do the try catch here?
useQuery returns an error, just return that

`const {
      data = {},
      error,
      isLoading,
      isSuccess,
    } = useQuery`

File: components/checkout/ShippingStep/ShippingStep.tsx [hunk 5]
```diff
@@ -96,6 +97,9 @@ const ShippingStep = (props: ShippingProps) => {
   const handleFormStatusChange = (status: boolean) => {
     setIsAddressFormValid(status)
   }
+  const handleAddressSelect = (_addressId: string) => {
+    // do your stuff for the selected address
```
Comment (kibo-kevinwatts) [inline components/checkout/ShippingStep/ShippingStep.tsx:101]: shouldn't this be implemented in this pull request ?

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: components/checkout/ShippingStep/ShippingStep.tsx (hunk 1 above)
Comment (sss-bct) [reply components/checkout/ShippingStep/ShippingStep.tsx]: Resolved.

File: components/common/AddressList/AddressList.stories.tsx (hunk 2 above)
Comment (sss-bct) [reply components/common/AddressList/AddressList.stories.tsx]: Resolved.

File: components/common/AddressList/AddressList.tsx (hunk 3 above)
Comment (sss-bct) [reply components/common/AddressList/AddressList.tsx]: Correct. Not required here.

File: components/checkout/ShippingStep/ShippingStep.tsx [hunk 6]
```diff
@@ -96,6 +97,25 @@ const ShippingStep = (props: ShippingProps) => {
   const handleFormStatusChange = (status: boolean) => {
     setIsAddressFormValid(status)
   }
+  const handleAddressSelect = (addressId: string) => {
+    const selectedAddress = userShippingAddress?.find((address) => address.id === Number(addressId))
+    if (selectedAddress?.id) {
+      const contact: Contact = {
+        id: selectedAddress.id,
+        firstName: selectedAddress.firstName || '',
```
Comment (kibo-kevinwatts) [inline components/checkout/ShippingStep/ShippingStep.tsx:105]: in addition to sending `id: selectedAddress.id`
add `contactId: selectedAddress.id`

File: lib/gql/fragments/userContact.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
+export const userContactFields = /* GraphQL */ `
+  fragment userContactFields on CustomerContact {
+    accountId
+    types {
+      name
+      isPrimary
+    }
+    auditInfo {
```
Comment (kibo-kevinwatts) [inline lib/gql/fragments/userContact.ts:8]: remove auditInfo, faxNumber, label, companyOrOrganization

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: lib/gql/fragments/userContact.ts (hunk 7 above)
Comment (sss-bct) [reply lib/gql/fragments/userContact.ts]: Removed.

File: components/checkout/ShippingStep/ShippingStep.tsx (hunk 6 above)
Comment (sss-bct) [reply components/checkout/ShippingStep/ShippingStep.tsx:105]: Type support is not there for contactId, should I update type for this?

File: components/page-templates/Checkout/Checkout.tsx [hunk 8]
```diff
@@ -42,6 +43,11 @@ const Checkout = (props: CheckoutProps) => {
     initialCheckout,
   })
 
+  const { user } = useAuthContext()
+  const { data: savedUserAddressData } = useUserAddressesQueries({
+    accountId: user?.id as number, // geetanshu
```
Comment (kibo-sushant) [inline components/page-templates/Checkout/Checkout.tsx:48]: remove commented code

File: __mocks__/stories/userAddressMock.ts [hunk 9]
```diff
@@ -0,0 +1,94 @@
+import { CustomerContactCollection } from '@/lib/gql/types'
+
+export const userAddressResponse: CustomerContactCollection = {
```
[the remark this answers is not in the document]
Comment (sss-bct) [reply __mocks__/stories/userAddressMock.ts:3]: it's there with another export here

File: components/common/AddressList/AddressList.tsx [hunk 10]
```diff
@@ -0,0 +1,141 @@
+import React, { useEffect, useState } from 'react'
+
+import { Box, Stack, Typography } from '@mui/material'
+import { useTranslation } from 'next-i18next'
+
+import AddressCard from '@/components/common/AddressCard/AddressCard'
+import KiboRadio from '@/components/common/KiboRadio/KiboRadio'
```
[the remark this answers is not in the document]
Comment (sss-bct) [reply components/common/AddressList/AddressList.tsx]: Resolved.

File: components/common/AddressList/AddressList.tsx [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+
+import type { CrAddress, CuAddress, CustomerContact } from '@/lib/gql/types'
+
+interface AddressListProps {
+  heading?: string
+  subHeading?: string
+  radio?: boolean
+  addresses: CustomerContact[] | undefined
+  selectedAddressId?: string
+  onAddressSelection: (addressId: string) => any
```
[the remark this answers is not in the document]
Comment (sss-bct) [reply components/common/AddressList/AddressList.tsx]: Done

File: components/common/AddressList/AddressList.tsx [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+interface KiboAddressListProps {
+  addresses: CustomerContact[] | undefined
+  heading?: string
+  subHeading?: string
+  selectedAddressId?: string
+  onAddressSelect: (addressId: string) => any
```
[the remark this answers is not in the document]
Comment (sss-bct) [reply components/common/AddressList/AddressList.tsx]: Done.

File: components/common/AddressList/AddressList.tsx [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+const buildAddressProps = (address: CrAddress | CuAddress | any) => {
```
[the remark this answers is not in the document]
Comment (sss-bct) [reply components/common/AddressList/AddressList.tsx]: Done.

File: components/common/AddressList/AddressList.spec.js [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+    expect(addressList).toHaveLength(addressCount)
+  })
+
+  it('should render the component with radio button when radio prop is true', () => {
+    render(<Radio {...Radio.args} />)
+    const addressListRadio = screen.getAllByTestId('address-card-list-radio-mock')
+    expect(addressListRadio[0]).toBeVisible()
+    expect(addressListRadio).toHaveLength(addressCount)
+  })
+})
```
[the remark this answers is not in the document]
Comment (sss-bct) [reply components/common/AddressList/AddressList.spec.js:32]: Handled in integration test.

File: components/page-templates/Checkout/Checkout.tsx (hunk 8 above)
Comment (sss-bct) [reply components/page-templates/Checkout/Checkout.tsx]: Done.

**Review by kibo-sushant — APPROVED** [review]

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**